protocol/communities_messenger_test.go

```diff
@@ -3242,3 +3242,78 @@ func (s *MessengerCommunitiesSuite) TestCommunityTokensMetadata() {
 	s.Require().Equal(tokensMetadata[0].Symbol, newToken.Symbol)
 	s.Require().Equal(tokensMetadata[0].Name, newToken.Name)
 }
+
+func (s *MessengerCommunitiesSuite) TestCommunityBanUserRequesToJoin() {
+	description := &requests.CreateCommunity{
+		Membership:  protobuf.CommunityPermissions_NO_MEMBERSHIP,
+		Name:        "status",
+		Color:       "#ffffff",
+		Description: "status community description",
+	}
+
+	// Create an community chat
+	response, err := s.bob.CreateCommunity(description, true)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+
+	community := response.Communities()[0]
+
+	response, err = s.bob.InviteUsersToCommunity(
+		&requests.InviteUsersToCommunity{
+			CommunityID: community.ID(),
+			Users:       []types.HexBytes{common.PubkeyToHexBytes(&s.alice.identity.PublicKey)},
+		},
+	)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+
+	community = response.Communities()[0]
+	s.Require().True(community.HasMember(&s.alice.identity.PublicKey))
+
+	response, err = s.bob.BanUserFromCommunity(
+		&requests.BanUserFromCommunity{
+			CommunityID: community.ID(),
+			User:        common.PubkeyToHexBytes(&s.alice.identity.PublicKey),
+		},
+	)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.Communities(), 1)
+
+	community = response.Communities()[0]
+	s.Require().False(community.HasMember(&s.alice.identity.PublicKey))
+	s.Require().True(community.IsBanned(&s.alice.identity.PublicKey))
+
+	// Retrieve request to join
+	err = tt.RetryWithBackOff(func() error {
```

Comment:
This is the issue, you need to make sure the community is retrieved, so within the backoff function, you will have to check `response`, you should be use `WaitOnMessengerResponse` and have your condition inside the function